service/src/main/kotlin/fi/espoo/evaka/invoicing/data/VoucherValueDecisionQueries.kt

```diff
@@ -331,12 +355,136 @@ fun Handle.searchValueDecisions(
+            partner.first_name as partner_first_name,
+            partner.last_name as partner_last_name,
+            partner.social_security_number as partner_ssn,
+            partner.street_address as partner_street_address,
+            partner.postal_code as partner_postal_code,
+            partner.post_office as partner_post_office,
+            partner.restricted_details_enabled as partner_restricted_details_enabled,
+            approved_by.first_name as approved_by_first_name,
+            approved_by.last_name as approved_by_last_name,
+            child.first_name as child_first_name,
+            child.last_name as child_last_name,
+            child.social_security_number as child_ssn,
+            child.street_address as child_address,
+            child.postal_code as child_postal_code,
+            child.post_office as child_post_office,
+            child.restricted_details_enabled as child_restricted_details_enabled,
+            daycare.name as placement_unit_name,
+            daycare.language as placement_unit_lang,
+            care_area.id as placement_unit_area_id,
+            care_area.name as placement_unit_area_name
+        FROM voucher_value_decision as decision
+            LEFT JOIN voucher_value_decision_part as part ON decision.id = part.voucher_value_decision_id
+            LEFT JOIN person as head ON decision.head_of_family = head.id
+            LEFT JOIN person as partner ON decision.partner = partner.id
+            LEFT JOIN person as child ON part.child = child.id
+            LEFT JOIN daycare ON part.placement_unit = daycare.id
+            LEFT JOIN care_area ON daycare.care_area_id = care_area.id
+            LEFT JOIN employee as approved_by ON decision.approved_by = approved_by.id
+        WHERE decision.id = :id
+        ORDER BY part.date_of_birth DESC
+    """
+
+    return createQuery(sql)
+        .bind("id", id)
+        .map(toVoucherValueDecisionDetailed(mapper))
+        .let { it.merge() }
+        .firstOrNull()
+}
+
+fun Handle.approveValueDecisionDraftsForSending(ids: List<UUID>, approvedBy: UUID) {
+    // language=sql
+    val sql =
+        """
+        UPDATE voucher_value_decision SET
+            status = :status,
+            decision_number = nextval('voucher_value_decision_number_sequence'),
+            approved_by = :approvedBy,
+            approved_at = NOW()
+        WHERE id = :id
+        """.trimIndent()
+
+    val batch = prepareBatch(sql)
+    ids.forEach { id ->
+        batch
+            .bind("status", VoucherValueDecisionStatus.WAITING_FOR_SENDING)
+            .bind("approvedBy", approvedBy)
+            .bind("id", id)
+            .add()
+    }
+    batch.execute()
+}
+
+fun Handle.getVoucherValueDecisionDocumentKey(id: UUID): String? {
+    // language=sql
+    val sql = "SELECT document_key FROM voucher_value_decision WHERE id = :id"
+
+    return createQuery(sql)
+        .bind("id", id)
+        .mapTo<String>()
+        .firstOrNull()
```

Comment:
Personal preference: if you're expecting only one result, prefer `singleOrNull`

Reply:
Can you elaborate why? Because `first` implies that there are more elements?

Reply:
Nevermind, I read the function's java doc.
> Returns single element, or `null` if the list is empty or has more than one element.

Reply:
Actually, it would be nicer if it threw an exception when the list has more than one element... 😞

Reply:
TIL: `singleOrNull` returns null when there's more than one :cry: I thought it would throw an exception (like `single` does for more than one element cases)

Reply:
The original point was to enforce the assumption "0-1 elements", but it doesn't matter whether you use `firstOrNull` or `singleOrNull` since neither does exactly the right thing